core/statedb/statedb.go

```diff
@@ -193,6 +193,50 @@ func (s *StateDB) GetAccount(accountIndex int64) (*account.Account, error) {
 	return account, nil
 }
 
+func (s *StateDB) GetAccountByName(accountName string) (*account.Account, error) {
+	for accountIndex := range s.PendingAccountMap {
```

Comment:
`s.PendingAccountMap` is same as `s.StateCache.PendingAccountMap`, so `exist` must be true. We can remove `GetPendingAccount` call

Reply:
it seems the call is to query the account from the cache, we can not remove the call if we want to implement this function in this way

Reply:
Yeah, I mean changing from `for accountIndex := range s.PendingAccountMap` to `for accountIndex, account := range s.PendingAccountMap`

Reply:
Fixed, thanks.